components/kyma-environment-broker/internal/process/input/input.go

```diff
@@ -686,6 +674,73 @@ func (r *RuntimeInput) setNodesForTrialUpgrade() error {
 	return nil
 }
 
+func (r *RuntimeInput) setOIDCForProvisioning() *gqlschema.OIDCConfigInput {
+	oidcConfig := &gqlschema.OIDCConfigInput{
+		ClientID:       r.oidcDefaultValues.ClientID,
+		GroupsClaim:    r.oidcDefaultValues.GroupsClaim,
+		IssuerURL:      r.oidcDefaultValues.IssuerURL,
+		SigningAlgs:    r.oidcDefaultValues.SigningAlgs,
+		UsernameClaim:  r.oidcDefaultValues.UsernameClaim,
+		UsernamePrefix: r.oidcDefaultValues.UsernamePrefix,
+	}
+
+	if r.provisioningParameters.Parameters.OIDC.IsProvided() {
+		r.setOIDCFromProvisioningParameters(oidcConfig)
+	}
+
+	return oidcConfig
+}
+
+func (r *RuntimeInput) setOIDCForUpgrade() *gqlschema.OIDCConfigInput {
+	oidcConfig := r.oidcLastValues
+	r.setOIDCDefaultValuesIfEmpty(&oidcConfig)
```

Comment:
nit: curious if this is for some backward compatibility. Because if I'm interpreting the workflow correctly, upgrade will only happen after provisioning and provisioning already sets defaults, so technically there shouldn't be any empty values I guess.